corehq/apps/data_interfaces/utils.py

```diff
@@ -34,3 +35,32 @@ def add_cases_to_case_group(domain, case_group_id, uploaded_data):
         case_group.save()
 
     return response
+
+
+def archive_forms(user, uploaded_data):
+    response = {
+        'errors': [],
+        'success': [],
+    }
+
+    for row in uploaded_data:
+        xform_id = row.get('form_id')
+        try:
+            xform = XFormInstance.get(xform_id)
+        except Exception as e:
+            response['errors'].append(
+                _(u"Could not get XFormInstance {}: {}").format(xform_id, e))
+            continue
+
+        xform_string = _(u"XFORM {} for domain {} by user '{}'").format(
+            xform['_id'],
+            xform['domain'],
+            user.username)
```

Comment:
Should have mentioned this before (and no need to change it now). It's nice to use named placeholders since they can be rearranged. This allows other languages to use different word order:
```python
xform_string = _(u"XFORM {form_id} for domain {domain} by user '{username}'").format(
    form_id=xform['_id'],
    domain=xform['domain'],
    username=user.username,
)
```